lib/unifex/unifex_cnode.ex

```diff
@@ -0,0 +1,168 @@
+defmodule Unifex.UnifexCNode do
+  @doc """
+  Wraps Bundlex.CNode functionalities, in due to support specific Unifex's CNode behaviours
+  """
+
+  require Bundlex.CNode
+
+  @enforce_keys [:server, :node]
+  defstruct @enforce_keys
+
+  @type t :: %__MODULE__{
+          server: pid,
+          node: node
+        }
+
+  @type on_start_t :: {:ok, t} | {:error, :spawn_cnode | :connect_to_cnode}
+
+  def cast_on_start_t({:ok, %Bundlex.CNode{} = bundex_cnode}) do
+    {:ok, cast_cnode(bundex_cnode)}
+  end
+
+  def cast_on_start_t({:ok, %__MODULE__{} = unifex_cnode}) do
+    {:ok, cast_cnode(unifex_cnode)}
+  end
+
+  def cast_on_start_t(on_start) do
+    on_start
+  end
+
+  def cast_cnode(%Bundlex.CNode{server: server, node: node}) do
+    %__MODULE__{
+      server: server,
+      node: node
+    }
+  end
+
+  def cast_cnode(%__MODULE__{server: server, node: node}) do
+    %Bundlex.CNode{
+      server: server,
+      node: node
+    }
+  end
+
+  @doc """
+  Calls Bundlex.CNode functions.
+  Look at Bundlex.CNode docs, to see more.
```

Comment:
This says nothing :P btw, read https://hexdocs.pm/elixir/writing-documentation.html, especially 'recommendations' section